tracsubtickets/api.py

```diff
@@ -91,44 +91,51 @@ def environment_needs_upgrade(self, db):
 
         return False
 
-    def upgrade_environment(self, db):
+    def upgrade_environment(self, db=None):
         db_manager, _ = DatabaseManager(self.env)._get_connector()
 
         # update the version
-        old_data = {} # {table.name: (cols, rows)}
-        cursor = db.cursor()
-        if not self.found_db_version:
-            cursor.execute("INSERT INTO system (name, value) VALUES (%s, %s)",
-                           (db_default.name, db_default.version))
-        else:
-            cursor.execute("UPDATE system SET value=%s WHERE name=%s",
-                           (db_default.version, db_default.name))
+        with self.env.db_transaction as db:
```

Comment:
One other thing about Trac versions. Officially Trac 1.0 still supports Python 2.5. That's a rather old Python version, but if you do care about it, you'll need to add `from __future__ import with_statement` to the top of any files in which a transaction context manager is used.